### DISCUSSION TITLE
Can I use real dataset for GPT2-gemini training?

### DISCUSSION THREAD

**Discussion by yurishin929** [hpcaitech/ColossalAI#2649] [Community | Q&A] [answered]
Hi, I want to use real dataset for gpt2 gemini training.
There's code for dummy dataset only. but GPT-Titans has the code processing real dataset([colossal-ai sample dataset](https://github.com/hpcaitech/ColossalAI/tree/main/examples/language/gpt/titans#how-to-prepare-webtext-dataset))

so can I use wikitext or sample dataset like titans? If I can, can I use [this code](https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/gpt/titans/dataset/webtext.py)  ?

I need I need to change gemini/commons/utils.py code in order to use other datasets, but I'm new to preprocess language data... thank you!
[https://github.com/hpcaitech/ColossalAI/tree/main/examples/language/gpt/gemini](url)

**Comment by feifeibear** [comment, marked as answer]
Hi, of course, you can use a real dataset for training. The code under the example directory is to demonstrate how to use the Gemini in your applications. Because how to load a dataset is usually a personal thing, we use dummy data here.

You can refer the the following instructions to prepare a Webtext dataset.
https://github.com/hpcaitech/ColossalAI-Examples/blob/main/language/gpt/README.md#how-to-prepare-webtext-dataset

**Reply by yurishin929** [reply]
Thank you for your reply.
but it seems that there's only the example for Titans in the instruction you mentioned.
how can I use webtext with [train_gpt_demo.py](https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/gpt/gemini/train_gpt_demo.py#L299)?
It requires input_ids and attn_mask from datasets directly, but WebtextDataset just returns train_ds(in the following code).
It can make train_dataloader from train_ds , but i don't know how to get input_ids, attn_mask from dataloader.
https://github.com/hpcaitech/ColossalAI/blob/a020eecc7051083e1dbc4a02bd49a9521b032aad/examples/language/gpt/titans/train_gpt.py#L46
Also, I tried to install the packages for LSH with `python setup.py install`,  but It failed with `error: command '/usr/bin/gcc' failed with exit code 1` even though I replaced the cMinhash.cpp in LSH/lsh with tools/lsh/cMinhash.cpp.
my gcc version is 9.4.0.
Is this the cause?

**Reply by yurishin929** [reply]
or Instead of training loop like[ this example](https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/gpt/gemini/train_gpt_demo.py#L297), can I use [trainer module](https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/gpt/titans/train_gpt.py#L104) for gemini?

**Reply by feifeibear** [reply]
of course, you can use the `fit`, which is a high-level API.

**Reply by yurishin929** [reply]
Thank you so much. I'll try it.

**Reply by yurishin929** [reply]
by the way, is this a typo? **2022**0105.
https://github.com/hpcaitech/ColossalAI-Examples#20220105

**Reply by feifeibear** [reply]
yes, it should 2023.

**Reply by binmakeswell** [reply]
> by the way, is this a typo? **2022**0105. https://github.com/hpcaitech/ColossalAI-Examples#20220105

Thanks, this typo has been fixed.

**Comment by yangtailu** [comment]
为什么我这用不了呢？

**Reply by binmakeswell** [reply]
> 为什么我这用不了呢？

Hi @yangtailu You can open a new issue and share more details to us. Thanks.
